src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java

```diff
@@ -525,7 +544,48 @@ public Object handle(
             {
                 BuildingWareHouse.handleBuildingOverChest(pos, (TileEntityChest) entity, world);
             }
-            else if (!world.setBlockState(pos, blockState, 0x03))
+            else if (!world.setBlockState(pos, blockState, UPDATE_FLAG))
+            {
+                return ActionProcessingResult.DENY;
+            }
+
+            return blockState;
+        }
+    }
+
+    public static class RackPlacementHandler implements IPlacementHandler
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 36 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2972)